test/integration/serviceaccount/service_account_test.go

```diff
@@ -573,58 +479,15 @@ func createServiceAccountToken(c clientset.Interface, sa *v1.ServiceAccount, ns
 	return secret, nil
 }
 
-func getReferencedServiceAccountToken(c clientset.Interface, ns string, name string, shouldWait bool) (string, string, error) {
-	tokenName := ""
-	token := ""
-
-	findToken := func() (bool, error) {
-		user, err := c.CoreV1().ServiceAccounts(ns).Get(context.TODO(), name, metav1.GetOptions{})
-		if apierrors.IsNotFound(err) {
-			return false, nil
-		}
-		if err != nil {
-			return false, err
-		}
-
-		for _, ref := range user.Secrets {
-			secret, err := c.CoreV1().Secrets(ns).Get(context.TODO(), ref.Name, metav1.GetOptions{})
-			if apierrors.IsNotFound(err) {
-				continue
-			}
-			if err != nil {
-				return false, err
-			}
-			if secret.Type != v1.SecretTypeServiceAccountToken {
-				continue
-			}
-			name := secret.Annotations[v1.ServiceAccountNameKey]
-			uid := secret.Annotations[v1.ServiceAccountUIDKey]
-			tokenData := secret.Data[v1.ServiceAccountTokenKey]
-			if name == user.Name && uid == string(user.UID) && len(tokenData) > 0 {
-				tokenName = secret.Name
-				token = string(tokenData)
-				return true, nil
-			}
-		}
-
-		return false, nil
+func addReferencedServiceAccountToken(c clientset.Interface, ns string, name string, secret *v1.Secret, shouldWait bool) error {
```

Comment:
in an integration test where we know there are no other controllers running, it would probably be simpler to get/modify/update the service account and fail if there's any error, including a conflict error

Reply:
also, I don't think the `shouldWait` parameter is needed

Reply:
done